dnf5-plugins/config-manager_plugin/unsetopt.cpp

```diff
@@ -107,48 +114,88 @@ void ConfigManagerUnsetOptCommand::configure() {
     const auto & config = ctx.base.get_config();
 
     // Remove options from main configuration file.
-    const auto & cfg_filepath = get_config_file_path(ctx.base.get_config());
-    if (!main_opts_to_remove.empty() && std::filesystem::exists(cfg_filepath)) {
-        ConfigParser parser;
-        bool changed = false;
-
-        parser.read(cfg_filepath);
-
-        changed |= remove_from_config(parser, "main", main_opts_to_remove);
+    if (!main_opts_to_remove.empty()) {
+        const auto & cfg_filepath = get_config_file_path(ctx.base.get_config());
+        if (std::filesystem::exists(cfg_filepath)) {
+            ConfigParser parser;
+            bool changed = false;
+            std::set<std::string> used_keys;
+
+            parser.read(cfg_filepath);
+
+            changed |= remove_from_config(parser, "main", main_opts_to_remove, used_keys);
+
+            // Generate warning for unused options
+            for (const auto & key : main_opts_to_remove) {
+                if (!used_keys.contains(key)) {
+                    ctx.base.get_logger()->warning(
+                        "config-manager: Request to remove not present main option from config file: {}", key);
```

Comment:
Just one small thing - this seems a bit hard to read. What about to unite it with the other warnings to e.g.: "Request to remove main option but it is not present in the config file:"